### PR TITLE
init impl

### LINKED ISSUE #192: Update delete workflow - side nav is not updated correctly
1. Delete workflow - When Project contains only one workflow and that workflow is deleted, it shows empty project in side navigation. - Project should be removed too.

2. Update workflow - When Project contains only one workflow and that workflow is updated with new project name, it shows old project name in side navigation. - Project should be updated too.

### PR DESCRIPTION
**Pull request by senelesithole**
fix #192

### CODE REVIEW

**Review by kevinwallimann — CHANGES_REQUESTED** [review]
Imo, this doesn't address the root cause of the bug. The problem is probably in the Workflow Reducer, e.g. in the case `WorkflowsActions.DELETE_WORKFLOW_SUCCESS` you can see that old projects are never deleted from the store, unless the whole page is refreshed ofc.
You can see in the screenshot (from redux dev tools) that the rename of the project simply lead to an addition of a new project instance in the store.
image
I guess the underlying confusion is because in the data model, `workflowModel.project` is thought to be merely a reference to project, while the definition of the project name is in `ProjectModel.name`. However in the UI, the Project Name input field simultaneously serves as both reference and definition of a project name. Honestly I think we haven't done ourselves a favour to not cleanly separate project definition from usage, but now we have to live with it.

File: ui/src/app/stores/workflows/workflows.reducers.ts [hunk 1]
```diff
@@ -245,12 +245,13 @@ export function workflowsReducer(state: State = initialState, action: WorkflowsA
         loading: true,
```
Comment (jozefbakus) [inline ui/src/app/stores/workflows/workflows.reducers.ts:245]: Cool

**Review by jozefbakus — APPROVED** [review]

**Review by kevinwallimann — APPROVED** [review]
That was easier than I expected 👌

**Comment by jozefbakus** [comment]
> That was easier than I expected 👌

I thought same thing :D

**Comment by senelesithole** [comment]
The lines of code don't really reflect the complexity of the issue 😊.

so that wasn't easy.

Thanks, Guys.

**Pull request merged**